### PR TITLE
feat(auth): Refresh hosted ui user pool token

### PR DESCRIPTION
**Pull request by tylerjroach**
- [ ] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*

*Description of changes:*

*How did you test these changes?*
(Please add a line here how the changes were tested)

- [ ] Added Unit Tests
- [ ] Added Integration Tests

*Documentation update required?*
- [ ] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sktimalsina — COMMENTED** [review]
LGTM, make sure to add follow up item for `//TODO: Better Exception`

**Comment by tylerjroach** [comment]
> LGTM, make sure to add follow up item for `//TODO: Better Exception`

Yes, next ticket is to overhaul hosted ui exceptions so I purposefully added that to be part of overhaul.

**Review by sktimalsina — APPROVED** [review]

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/RefreshSessionState.kt [hunk 1]
```diff
@@ -81,9 +82,16 @@ sealed class RefreshSessionState : State {
             return when (oldState) {
                 is NotStarted -> when (refreshSessionEvent) {
                     is RefreshSessionEvent.EventType.RefreshUserPoolTokens -> {
-                        val action = fetchAuthSessionActions.refreshUserPoolTokensAction(
-                            refreshSessionEvent.signedInData
-                        )
+                        val action = when (refreshSessionEvent.signedInData.signInMethod) {
+                            is SignInMethod.HostedUI -> {
```
Comment (div5yesh) [inline aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/RefreshSessionState.kt:86]: 🙌

**Review by div5yesh — APPROVED** [review]
Ship it! 🚀

**Pull request merged**